### PR TITLE
refactor: extract DrawerHeader component

### PR DESCRIPTION
**Pull request by karrui**
### TL;DR

Introduced a new DrawerHeader component and refactored existing drawer components to use it.

### What changed?

- Created a new `DrawerHeader` component in `Drawer/DrawerHeader.tsx`
- Refactored `ComplexEditorStateDrawer`, `HeroEditorDrawer`, `MetadataEditorStateDrawer`, and `TipTapProseComponent` to use the new `DrawerHeader`
- Removed redundant imports and simplified code in the refactored components
- Replaced lodash imports with specific function imports to optimize bundle size

### How to test?

1. Open the editing experience for various components (Complex, Hero, Metadata, and Prose)
2. Verify that the drawer header appears consistent across all components
3. Test the back button functionality in each drawer
4. Ensure that the discard changes modal appears when appropriate

### Why make this change?

This change improves code consistency and maintainability by centralizing the drawer header logic into a single component. It also reduces code duplication and improves performance by optimizing imports. The new `DrawerHeader` component provides a unified look and feel across different editing experiences, enhancing the overall user interface consistency.

### CODE REVIEW

**Comment by karrui** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/603?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#603** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/603?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#599** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/599?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#592** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/592?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by seaerchin — APPROVED** [review]

File: apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx [hunk 1]
```diff
@@ -246,46 +237,17 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
...
-        </Box>
-
+        <DrawerHeader
+          isDisabled={isLoading}
+          onBackClick={() => {
+            if (!isEqual(previewPageState, savedPageState)) {
+              onDiscardChangesModalOpen()
+            } else {
+              handleDiscardChanges()
+            }
```
Comment (dcshzj) [inline apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx:247]: issue: this discard changes modal doesn't open when the user is in an invalid state (enter a block, remove title that is required, click back should show this modal)

Comment (karrui) [reply apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx:247]: this one fix on new issue ba

**Comment by karrui** [comment]
### Merge activity

* **Sep 12, 6:06 AM EDT**: @karrui started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/isomer/603).

**Pull request merged**